File: pset6/vigenere.py
import cs50
import sys

def main():
    # control for appropriate input
    if len(sys.argv) != 2:
        print("Inappropriate command line arguments: requires two arguments")
        exit(1)
    if sys.argv[1].isdigit():
        print("Inappropriate command line argument: no digits allowed")
        exit(2)
    # normal program operation
    else:
        p = input("Enter plaintext to encipher: ")
        encipher(p)
        
def encipher(p):
    k = sys.argv[1]
    j = 0
    alphaoffset = 26
    print("Ciphertext: ", end = "")
    # iterate through plaintext input
    for c in range(len(p)):
        # separate non-alphabet characters
        if p[c].isalpha():
            if p[c].isupper():
                # loop through cipher argument
                ml = j%len(k)
                # normalize array value to current case
                uc = k[ml].upper()
                # find alphavalue for cipher key value
                cc = ord(uc) - ord("A")
                # find alphavalue for plaintext value
                cb = ord(p[c]) - ord("A")
                # find offset value
                cp = (cc + cb)%alphaoffset
                # apply offset and print
                cf = cp + ord("A")
                print("{}".format(chr(cf)), end = "")
                j += 1
            if p[c].islower():       
                # loop through cipher argument
                ml = j%len(k)
                # normalize array value to current case
                uc = k[ml].lower()
                # find alphavalue for cipher key value
                cc = ord(uc) - ord("a")
                # find alphavalue for plaintext value
                cb = ord(p[c]) - ord("a")
                # find offset value
                cp = (cc + cb)%alphaoffset
                # apply offset and print
                cf = cp + ord("a")
                print("{}".format(chr(cf)), end = "")
                j += 1       
        else:
            print(p[c], end = "")
            
    print("")
        
if __name__ == "__main__":
    main()

File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int main (int argc, string argv[])
{
    //limit to two command line values
    if (argc != 2)
    {
        printf("Inappropriate command line arguments\n");
        return 1;
    }
    //fetch command line input and store length
    string k = argv[1];
    int lena = strlen(k);
    //limit command line value to text
    int g = 0;
    for (int f = 0; f<strlen(k); f++)
    {
        if(isdigit(k[f])){
         g++;   
        }
    }
    if (g>0)
    {
        printf("Inappropriate command line arguments\n");
        return 1;
    }
    //normal program operation
    else 
    {
        //collect plaintext input and store length
        printf("plaintext: ");
        string p = get_string();
        int lens = strlen(p);
        printf("ciphertext: ");
        //array control
        int j = 0;
        //iterate through plaintext input
        for (int i = 0; i < lens; i++)
        {
            //separate non-alphabet characters
            if (isalpha(p[i]))
                    {
                    //separate upper-case characters
                    if (isupper(p[i]))
                        {
                        //iterate through array values and choose appropriate
                        int ml = j%lena;
                        //normalize array value to current case
                        char uc = toupper(k[ml]);
                        //find alphavalue for key value
                        int c = (uc)-65;
                        //find alphavalue for plaintext value
                        int cb = p[i]-65;
                        //find outset amount
                        int cp = ((c+cb)%26);
                        //apply offset to character and print
                        int cc = cp+65;
                        printf("%c",cc);
                        j++;
                        }
            
                    if (islower(p[i]))
                        {
                        //iterate through array values and choose appropriate
                        int ml = j%lena;
                        //normalize array value to current case
                        char lc = tolower(k[ml]);
                        //find alphavalue for key value
                        int c = (lc)-97;
                        //find alphavalue for plaintext value
                        int cb = p[i]-97;
                        //find outset amount
                        int cp = ((c+cb)%26);
                        //apply offset to character and print
                        int cc = cp+97;
                        printf("%c",cc);
                        j++;
                        }
                    }
            else
            {
                printf("%c",p[i]);
            }
        }
    printf("\n");
    }
}
